### PR TITLE
Set v7 protocol as current version in docs

### PR DESCRIPTION
**Pull request by marioizquierdo**
Set the v7 protocol as the current version. Include differences with v5 so we can link other implementations to this doc. This should be merged along with a new release that enumerates all the changes that were included in this release:

 * #271
 * #270 
 * #268
 * #264
 * #257

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: draft_release_notes.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+A major release that includes a protocol spec update. The jump from v5 to v7 is to avoid confusion with issues and comments that were targeting v6 and were never released (see [achieved protocol spec v6](https://twitchtv.github.io/twirp/docs/spec_v6.html)).
```
Comment (amcohen-twitch) [inline draft_release_notes.md:1]: achieved => archived?

File: draft_release_notes.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+Changes in the Protocol spec v7:
+
+ * Twirp routes can have any prefix: `<prefix>/<package>.<Service>/<Method>` section. In v5 the prefix was mandatory was always `/twirp`.
```
Comment (amcohen-twitch) [inline draft_release_notes.md:5]: was mandatory was always => was mandatory, always

File: draft_release_notes.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ * Error code `ResourceExhausted` maps to HTTP code `429`. In v5 it mapped to `403`.
+
+Changes included in the Go Twirp v7 release:
+
+ * #264 Optional Twirp Prefix. Implements the proposal #263 with the optional prefix, using an option to specify a different prefix than the default value "/twirp". The default value ensures backwards compatibility when updating the service. Using different prefixes, it is now possible to mount the same Twirp service in multiple routes, which may help migrating existing services from the "/twirp" prefix to a new one.
+ * #264 also introduces [server options on the server constructor](https://github.com/twitchtv/twirp/pull/264#issuecomment-686170407). Adding server hooks should now done through a server option. The server options are available in the new version of the `twirp` package, servers generated with the new version require the `twirp` package to be updated as well.
+ * #270 ResourceExhausted error code to HTTP 429 status code. To implement the new spec for this error, so it is clear that this error code can be used for rate limiting. This change may affect middleware if it depends on a specific status code being used.
```
Comment (amcohen-twitch) [inline draft_release_notes.md:12]: > To implement the new spec for this error, so it is clear that this error code can be used for rate limiting.

This sentence doesn't quite make sense.

**Review by amcohen-twitch — CHANGES_REQUESTED** [review]

File: draft_release_notes.md (hunk 3 above)
Comment (marioizquierdo) [reply draft_release_notes.md]: 😄  what was I thinking? Updated. Thanks!

File: PROTOCOL.md [hunk 4]
```diff
@@ -145,30 +146,71 @@ Content-Length: 27
...
+| deadline_exceeded   | 408 | Operation expired before completion. For operations that change the state of the system, this error may be returned even if the operation has completed successfully (timeout).
+| not_found           | 404 | Some requested entity was not found.
+| bad_route           | 404 | The requested URL path wasn't routable to a Twirp service and method. This is returned by generated server code and should not be returned by application code (use "not_found" or "unimplemented" instead).
+| already_exists      | 409 | An attempt to create an entity failed because one already exists.
+| permission_denied   | 403 | The caller does not have permission to execute the specified operation. It must not be used if the caller cannot be identified (use "unauthenticated" instead).
+| unauthenticated     | 401 | The request does not have valid authentication credentials for the operation.
+| resource_exhausted  | 403 | Some resource has been exhausted, perhaps a per-user quota, or perhaps the entire file system is out of space.
```
Comment (thesilentg) [inline PROTOCOL.md:207]: Does this need to be updated to 429?

Comment (marioizquierdo) [reply PROTOCOL.md:207]: Yes, this entire file should be a copy-paste form the spec in the docs. Which I though I did already 🤔  good catch. I wonder if this should just contain a link to the docs ... but realistically it almost never changes, so it's fine to have some duplication 🤷

Comment (marioizquierdo) [reply PROTOCOL.md:207]: oh.. I see. I need to merge #270 first

**Pull request merged**